app/services/activities/import_from_csv.rb

```diff
@@ -87,11 +88,27 @@ def update
 
         @activity.assign_attributes(@converter.to_h)
 
+        associated_implementing_organisation = ImplementingOrganisation.find_or_initialize_by(
+          activity_id: @activity.id
+        ) { |implementing_organisation|
+          implementing_organisation.name = row["Implementing organisation name"]
+          implementing_organisation.reference = row["Implementing organisation reference"]
+          implementing_organisation.organisation_type = row["Implementing organisation sector"]
+        }
```

Comment:
I feel like this (and the lines in the creator) are similar enough to have as a class of their own? There's a fair bit of repetition in the error lines too, so was wondering if that stuff could be encapsulated in the class too?